hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -447,6 +449,8 @@ public RegisteredCommand register(
           processNodeReport(datanodeDetails, nodeReport);
           LOG.info("Updated datanode to: {}", dn);
           scmNodeEventPublisher.fireEvent(SCMEvents.NODE_ADDRESS_UPDATE, dn);
+        } else if (isVersionChange(oldNode.getVersion(), datanodeDetails.getVersion())) {
+          nodeStateManager.updateNode(datanodeDetails, layoutInfo);
```

Comment:
Thanks for the question!

We made this change to address the issue where SCM does not recognize different versions of the same UUID's DN. The version we changed is the `compilation` version of the DataNode, not the `layout` version.

For example, in our Ozone cluster with 10 DataNodes, the DN version is `1.4.0-202`. When these 10 DataNodes register with SCM, they all display as `1.4.0-202`, which is correct. However, if we upgrade one DataNode to version `1.4.0-203` and do not restart SCM, SCM will not recognize this new version number. This is because SCM does not update the DN version information during the heartbeat.

Reply:
I see, I think it should be fine. Could you update the related comment ("Update datanode if it is registered....") and add some INFO logs regarding the version change?

Reply:
Thank you for the suggestion! I will add some logs.

```
LOG.info("Update the version for registered datanode = {}, oldVersion = {}, newVersion = {}.",
        datanodeDetails.getUuid(), oldNode.getVersion(), datanodeDetails.getVersion());
```